components/src/core/components/List/types.ts

```diff
@@ -0,0 +1,126 @@
+interface Visible {
+  visible?: boolean;
+}
+
+interface ListEmitEvent {
+  type: string;
+  identifier: string;
+}
+
+interface Tooltip {
+  tooltip?: string;
+  flow?: string;
+}
+
+interface Style {
+  [key: string]: string | undefined;
+}
+
+interface Icon {
+  iconName?: string;
+  "icon-size"?: string;
+  iconSize?: string;
+  iconStyle?: string;
+}
+
+interface ButtonIcon extends Icon {
+  "icon-name"?: string;
+}
+
+interface ListButtonProps extends Tooltip, ButtonIcon {
+  label?: string;
+  displayType?: string;
+  size?: string;
+  "display-type"?: string;
+}
+
+interface ListButton {
+  props: ListButtonProps;
+  events?: ListEmitEvent[];
+  class?: string;
+  style?: Style;
+}
+
+interface ListBulkAction extends ListButton, Visible {
+  conditional?: boolean;
+  type?: string;
+}
```

Comment:
Can we call this ListAction? since it's used for both bulk and single actions?

Reply:
Renamed with ListAction